client/src-js/ac.js

```diff
@@ -0,0 +1,159 @@
+const HISTORY_SIZE = 20
+const HISTORY_LIST_SIZE = 4
+const DB_ID = "search_history"
+const WRAPPER_CLASS = "search_class"
+const HOVER_COLOR = '#f4ada4'
+const HOVER_COLOR_CANCEL = '#dddddd'
+const BG_COLOR = '#fefefe'
+const PRIMARY = '#ef8374'
+
+function range(n) {
+  let res = []
+  for (let i=0; i<n; i++) {
+    res.push(i)
+  }
+  return res
+}
+
+function getHistory() {
+  try {
+    return JSON.parse(localStorage.getItem(DB_ID))
+  } catch (e) {
+    return []
+  }
+}
+
+function deleteEntry(idx) {
+  const history = getHistory()
+  history.splice(idx, 1)
+  localStorage.setItem(DB_ID, JSON.stringify(history))
+}
+
+function hideMenu(n) {
+  const e = n.parentNode.querySelector('.'+WRAPPER_CLASS)
+  e && e.remove()
+}
+
+function onSearchKey(e) {
+  const ip = e.target
```

Comment:
Not sure why you call event target `ip`? Is it supposed to be `input`? If that's the case I would simply use `input`. Also relevant in other event handlers